context/app/static/js/components/search/Facets/DateRangeFacet.tsx

```diff
@@ -0,0 +1,164 @@
+import React, { useCallback, useState, useEffect } from 'react';
+import Stack from '@mui/material/Stack';
+import { DatePicker, DatePickerProps } from '@mui/x-date-pickers/DatePicker';
+import { DateValidationError } from '@mui/x-date-pickers/models';
+import { format } from 'date-fns/format';
+
+import { useSearch } from '../Search';
+import { isDateFilter, useSearchStore } from '../store';
+import { getFieldLabel } from '../fieldConfigurations';
+import FacetAccordion from './FacetAccordion';
+
+interface DateRangeFacetProps {
+  field: string;
+}
+
+function DatePickerComponent({
+  minDate,
+  maxDate,
+  ...rest
+}: DatePickerProps<Date> & Required<Pick<DatePickerProps<Date>, 'minDate' | 'maxDate'>>) {
+  const [error, setError] = React.useState<DateValidationError | null>(null);
+
+  const errorMessage = React.useMemo(() => {
```

Comment:
```suggestion
  const errorMessage = useMemo(() => {
```

It'll also need to be added to the import at the top.